app/code/Magento/ProductAlert/Block/Adminhtml/Catalog/Product/Edit/Tab/Alerts/Renderer/Stock.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Magento\ProductAlert\Block\Adminhtml\Catalog\Product\Edit\Tab\Alerts\Renderer;
+
+use Magento\Backend\Block\Widget\Grid\Column\Renderer\AbstractRenderer;
+use Magento\Framework\DataObject;
+use Magento\Framework\Message\ManagerInterface;
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\Backend\Block\Context;
+use Magento\InventorySales\Model\StockResolver;
+use Magento\InventorySalesApi\Api\Data\SalesChannelInterface;
+
+class Stock extends AbstractRenderer
+{
+    /**
+     * @var StoreManagerInterface
+     */
+    protected $_storeManager;
+
+    /**
+     * @var StockResolver
```

Comment:
Need to work only with abstraction (interfaces) but not with implementation

Reply:
Yes, I know, But I didn't find any prefence for StockResolverInterface. I've tried to use StockResolverInterface and I had an error.